domain/src/main/java/io/homeassistant/companion/android/domain/widgets/WidgetRepository.kt

```diff
@@ -0,0 +1,26 @@
+package io.homeassistant.companion.android.domain.widgets
+
+interface WidgetRepository {
+
+    suspend fun saveServiceCallData(
+        appWidgetId: Int,
+        domainStr: String,
+        serviceStr: String,
+        serviceDataStr: String
+    )
+
+    suspend fun loadDomain(appWidgetId: Int): String?
+    suspend fun loadService(appWidgetId: Int): String?
+    suspend fun loadServiceData(appWidgetId: Int): String?
+
+    suspend fun loadIcon(appWidgetId: Int): String?
+    suspend fun loadLabel(appWidgetId: Int): String?
+
+    suspend fun saveIcon(appWidgetId: Int, resName: String?)
+    suspend fun saveLabel(appWidgetId: Int, data: String?)
+
+    suspend fun deleteWidgetData(appWidgetId: Int)
+
+    suspend fun saveStringPref(key: String, data: String?)
+    suspend fun loadStringPref(key: String): String?
```

Comment:
Do these get called from anywhere besides WidgetRepositoryImpl.kt?  If they do we should setup another usecase for it rather than just exposing the local storage implementation.

Reply:
You mean inside the widget use case? Or a completely separate use case?

Reply:
I wasn't sure if this was called.  I just took a look and they don't need to be in the interface, they can be private functions in the impl.

Reply:
Okay, I'll change them to private in the WidgetRepositoryImpl when I get a chance later today.

Reply:
After that we'll get this merged!